datafusion/core/src/execution/context.rs

```diff
@@ -450,6 +457,36 @@ impl SessionContext {
         self.read_batch(record_batch)
     }
 
+    // Execute a COPY TO statement, returning the number of rows
```

Comment:
Note that this is the mechanism as used by `CREATE TABLE AS SELECT` (aka LogicalPlan::CreateMemTable`). It is different than the mechanism used by `INSERT INTO ... ` (added in https://github.com/apache/arrow-datafusion/pull/5520 by @metesynnada ) that uses an ExecutionPlan. 

The difference bothers me, but I can see the benefits of both approaches

Reply:
I was playing with this more this evening and I think I came up with something that is half way between that I like even better. Will keep iterating and report back

Reply:
I think we have something you may be able to leverage, check [this PR](https://github.com/synnada-ai/arrow-datafusion/pull/89) out. It extends the ExecutionPlan approach to writing files, I think you can leverage that work here too.

Reply:
Thanks @ozankabak  -- I have reviewed https://github.com/synnada-ai/arrow-datafusion/pull/89 and I have thought about how to incorporate the same structure

I really like your idea to use the the same plans for `COPY TO` and `INSERT INTO`. After some more thought, I have an idea of how to plan `COPY` statements using the same plans as an INSERT. 

Here is a proposal for a simplified API: https://github.com/apache/arrow-datafusion/issues/6339

I'll try and bang out a Proposal shortly